### PR TITLE
Fix vertex-centric-index multiplicity constraints

### PR DESCRIPTION
**Pull request by dpitera**
Issues: https://github.com/JanusGraph/janusgraph/issues/442

Signed-off-by: David Pitera <dpitera@us.ibm.com>

### CODE REVIEW

File: janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java [hunk 1]
```diff
@@ -1360,6 +1360,29 @@ public void testGotGIndexRemoval() throws InterruptedException, ExecutionExcepti
...
 
     @Test
+    public void testVertexCentricIndexOnSimpleMultiplicityShouldWork() {
+        clopen(option(LOG_SEND_DELAY, MANAGEMENT_LOG), Duration.ofMillis(0),
+                option(KCVSLog.LOG_READ_LAG_TIME, MANAGEMENT_LOG), Duration.ofMillis(50),
+                option(LOG_READ_INTERVAL, MANAGEMENT_LOG), Duration.ofMillis(250)
+        );
+        PropertyKey time = mgmt.makePropertyKey("time").dataType(String.class).cardinality(Cardinality.SINGLE).make();
+        EdgeLabel friend = mgmt.makeEdgeLabel("friend").multiplicity(Multiplicity.SIMPLE).make();
+        mgmt.buildEdgeIndex(friend, "byTime", Direction.OUT, decr, time);
```
Comment (pluradj) [inline janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java]: I think we'd need a test for `buildPropertyIndex()` too

Comment (dpitera) [inline janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java]: these indexes don't make usage of `edgeLabels` which deal with `MULTIPLICITY` concerns

**Comment by dpitera** [comment]
Thanks @pluradj. Updated code.

**Comment by krlohnes** [comment]
+1 (non-binding)

The difference between unique and constrained is a little subtle, but this is definitely correct. Looks to me like this was just a `method-name-looks-right` type of bug.

**Pull request merged**